dissect/target/plugin.py

```diff
@@ -951,7 +951,9 @@ def _filter_compatible(
     descriptors: list[FunctionDescriptor], target: Target, ignore_load_errors: bool = False
 ) -> Iterator[FunctionDescriptor]:
     """Filter a list of function descriptors based on compatibility with a target."""
-    seen = set()
+    compatible = set()
+    incompatible = set()
+
     for descriptor in descriptors:
         try:
             plugincls = load(descriptor)
```

Comment:
Maybe it's an idea to make the incompatible/compatible check on the descriptor? Then we can also skip loading it if we know it's already compatible or incompatible (or failed to load previously).